edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Scraper.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    public class Scraper : IMetricsScraper, IDisposable
```

Comment:
should it be PrometheusMetricScraper?

Reply:
please check out what should be done for implementing IDisposable, e.g. https://stackoverflow.com/questions/18336856/implementing-idisposable-correctly.

Reply:
?

I don't know what you mean here. I implemented ```Dispose```.